float64 delta_r1
float64 delta_trans
float64 delta_r2